format-parquet/src/main/java/io/cdap/plugin/format/parquet/input/ParquetInputFormatProvider.java

```diff
@@ -92,17 +93,28 @@ public Schema getSchema(FormatContext context) {
   public Schema getDefaultSchema(FormatContext context) throws IOException {
     String filePath = conf.getProperties().getProperties().getOrDefault("path", null);
     ParquetReader reader = null;
+    GenericData.Record record = null;
     try {
       Job job = JobUtils.createInstance();
       Configuration hconf = job.getConfiguration();
       // set entries here, before FileSystem is used
       for (Map.Entry<String, String> entry : conf.getFileSystemProperties().entrySet()) {
         hconf.set(entry.getKey(), entry.getValue());
       }
-      final Path file = conf.getFilePathForSchemaGeneration(filePath, ".+\\.parquet", hconf, job);
-      reader = AvroParquetReader.builder(file).build();
-      GenericData.Record record = (GenericData.Record) reader.read();
-      return Schema.parseJson(record.getSchema().toString());
+      List<Path> paths = conf.getFilePathsForSchemaGeneration(filePath, ".+\\.parquet$", hconf, job); 
+      for (Path file : paths) {
+        reader = AvroParquetReader.builder(file).build();
+        record = (GenericData.Record) reader.read();
+        if (record != null) {
+          return Schema.parseJson(record.getSchema().toString());
+        }
+        reader.close();
```

Comment:
Use try with resource to make sure resource is always closed (e.g. in this case, since you have a `return` earlier, the reader actually never get closed).